sdks/go/pkg/beam/external.go

```diff
@@ -16,10 +16,151 @@
 package beam
 
 import (
+	"context"
+	"fmt"
+
 	"github.com/apache/beam/sdks/go/pkg/beam/core/graph"
+	"github.com/apache/beam/sdks/go/pkg/beam/core/runtime/graphx"
 	"github.com/apache/beam/sdks/go/pkg/beam/internal/errors"
+	jobpb "github.com/apache/beam/sdks/go/pkg/beam/model/jobmanagement_v1"
+	pipepb "github.com/apache/beam/sdks/go/pkg/beam/model/pipeline_v1"
+	"google.golang.org/grpc"
 )
 
+// ExternalTransform represents the cross-language transform in and out of the Pipeline as a MultiEdge and Expanded proto respectively
+type ExternalTransform struct {
+	id                int
+	Urn               string
+	Payload           []byte
+	In                []PCollection
+	Out               []FullType
+	Bounded           bool
+	ExpansionAddr     string
+	Components        *pipepb.Components
+	ExpandedTransform *pipepb.PTransform
+	Requirements      []string
+}
+
+// CrossLanguage is the temporary API to execute external transforms
+// TODO(pskevin): Handle errors using the TryN and Must strategies instead one function handling multiple points of failure
+func CrossLanguage(s Scope, p *Pipeline, e *ExternalTransform) []PCollection {
+	if e.ExpansionAddr == "" { // TODO(pskevin): Better way to check if the value was ever set
+		// return Legacy External API
+	}
+
+	/*
+		Add ExternalTranform to the Graph
```

Comment:
I'm definitely missing something since I can't see any typo. Could you be more explicit?

Also, the block comments were meant just to separate sections of independent logic which I expect to break down into functions during clean up. Should I remove all the block comments?

Reply:
ExternalTranform should be ExternalTransform.

I'd prefer no block comments. They're uncommon in Go code.  

To be clear, commenting on what the next sections of code is what comments are for. They don't need to take up so much space though.

Reply:
To be pedantically clear: No block comments doesn't mean no comments. In my original comment, I suggested the line comment alternative.

Reply:
Thanks for clarifying in detail. Reflected the changes.